compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt

```diff
@@ -193,8 +159,20 @@ class ComposeScene internal constructor(
 
     private inline fun forEachOwner(action: (SkiaBasedOwner) -> Unit) {
         listCopy.addAll(owners)
-        listCopy.forEach(action)
-        listCopy.clear()
+        try {
+            listCopy.forEach(action)
+        } finally {
+            listCopy.clear()
+        }
+    }
+
+    private inline fun <T> reversedOwners(action: (List<SkiaBasedOwner>) -> T): T {
+        listCopy.addAll(owners)
```

Comment:
Do we need `listCopy` for performance reasons?
Otherwise, it's easier to use `owners.reversed()`. It will create a copy of the list under the hood. And will not change the original list of "owners".

Reply:
> Do we need listCopy for performance reasons?

It's not changed. Yes, it's for reducing allocations

Reply:
The original `forEachOwner` was added to avoid concurrent access to the list (an owner can be removed during the list iteration).

Reply:
> The original forEachOwner was added to avoid concurrent access to the list (an owner can be removed during the list iteration).

In this case, we can simply call `owners.reversed()` to make a copy and iterate over it.
Also, we may use function `owners.toList()` to create a copy and iterate in original order.

I think that usage of additional mutable `listCopy` is redundant here.
But it is not a call to action, it is just my thought.

Reply:
> owners.reversed(), owners.toList()

They both create a new list.

 `listCopy` was created to avoid allocations. It may unneccessary optimization, at least on JVM (it has young generation GC, which works very well with such allocations), but I wouldn't change the existed pattern right now.

If we all are not sure, we may add a comment to revisit it after we have benchmarks.